metadata-ingestion/src/datahub/configuration/config_loader.py

```diff
@@ -51,6 +52,19 @@ def resolve_env_variables(config: dict) -> dict:
     return new_dict
 
 
+def list_referenced_env_variables(config: dict) -> Set[str]:
```

Comment:
I don't see this function being used anywhere (other than the test). So what's the purpose of this function?

Reply:
If we want to fetch secrets from an external secret store rather than env vars, we'll need to be able to list the required variables. This function will be used by the acryl executor.